lstchain/reco/dl1_to_dl2.py

```diff
@@ -417,3 +429,68 @@ def apply_models(dl1, classifier, reg_energy, reg_disp_vector, custom_config={})
     dl2['gammaness'] = probs
     return dl2
 
+
+def set_source_dependent_parameters(data, config):
```

Comment:
This is a very large and complex (lot's of branching) function. Better split in multiple smaller functions for each of the cases you handle in the different if/else statements. This will also make it easier to write tests for this./